addons/repair/models/repair.py

```diff
@@ -317,7 +317,9 @@ def _compute_is_returned(self):
     @api.depends('move_ids.quantity', 'move_ids.product_uom_qty', 'move_ids.product_uom.rounding')
     def _compute_has_uncomplete_moves(self):
         for repair in self:
-            repair.has_uncomplete_moves = any(not move.picked or float_compare(move.quantity, move.product_uom_qty, precision_rounding=move.product_uom.rounding) < 0 for move in repair.move_ids)
+            def complete_move(move):
+                return not move.picked or float_compare(move.quantity, move.product_uom_qty, precision_rounding=move.product_uom.rounding) < 0
+            repair.has_uncomplete_moves = any(not complete_move(move) for move in repair.move_ids) and any(complete_move(move) for move in repair.move_ids)
```

Comment:
```suggestion
            incomplete_moves = repair.move_ids.filtered(lambda m: not m.picked or float_compare(m.quantity, m.product_uom_qty, precision_rounding=m.product_uom.rounding) < 0)
            repair.has_uncomplete_moves = incomplete_moves and len(incomplete_moves) != len(self.move_ids)
 ```
 Avoids some extra iterations